pagopa/io-app
# chore: [IOAPPCOM-42] Add mixpanel events for Premium

## Short description
This PR adds Premium mixpanel events (described [here](https://docs.google.com/spreadsheets/d/1bXNRcTDKU-5l2RqCcwSbT7Y1M9QQTWtS_f-3-AIqSYk/edit#gid=1055214768))

## List of changes proposed in this pull request
Following trackings have been added
- `NOTIFICATIONS_OPTIN_PREVIEW_STATUS` `{ enabled: true/false }` - Sent when the user taps on "Continue" in the Notifications Opt-In screen
- `NOTIFICATIONS_OPTIN_REMINDER_STATUS` `{ enabled: true/false }` - Sent when the user taps on "Continue" in the Notifications Opt-In screen
- `NOTIFICATIONS_OPTIN_REMINDER_ON_PERMISSIONS_OFF` - Sent when the user has either enabled notification previews or the reminders but has a disabled system notification permission
- `NOTIFICATIONS_OPTIN_OPEN_SETTINGS` - Sent when the user taps on the "Open Settings" button in the Info screen (that appears after the notification permissions has been denied, in the notification opt-in screen)
- `NOTIFICATIONS_OPTIN_SKIP_SYSTEM_PERMISSIONS` - Sent when the user taps on the "Not Now" button in the Info screen (that appears after the notification permissions has been denied, in the notification opt-in screen)
- `NOTIFICATIONS_PREFERENCES_PREVIEW_STATUS` `{ enabled: true/false }` - Sent when the user changes the preview preference, in the notification preference screen (profile section)
- `NOTIFICATIONS_PREFERENCES_REMINDER_STATUS` `{ enabled: true/false }` - Sent when the user changes the reminder preference, in the notification preference screen (profile section)
- `NOTIFICATIONS_MESSAGE_TAP { messageId: NonEmptyString }` - Sent when the user taps on a received notification. This event is used in a funnel with the `SESSION_EXPIRED` event (already in app) to track how many users will need to login again

## How to test
Using the dev server, check that every event above is triggered. To access the Opt-In screen, make sure that the profile configuration in the dev server has no `reminder_status` nor `push_notifications_content_type` properties set.
